src/helm/proxy/clients/openai_client.py

```diff
@@ -191,8 +193,83 @@ def do_it():
             embedding=embedding,
         )
 
+    @staticmethod
+    def _get_model_engine_name(model_engine: str) -> str:
+        return "/".join(model_engine.split("/")[1:])
+
     def tokenize(self, request: TokenizationRequest) -> TokenizationRequestResult:
-        raise NotImplementedError("Use the HuggingFaceClient to tokenize.")
+        # For reproducibility purposes, we use huggingface/gpt2 as the default tokenizer.
+        # except for gpt-3.5 turbo models and gpt-4 that use the tiktoken library.
+
+        if (
+            request.model_engine is not None
+            and self._get_model_engine_name(request.model_engine) not in OpenAIClient.MODELS_USING_TIKTOKEN
+        ):
+            return self.tokenizer_client.tokenize(
+                # We're assuming the model uses the GPT-2 tokenizer.
+                TokenizationRequest(request.text, tokenizer="huggingface/gpt2")
+            )
+
+        cache_key = asdict(request)
+
+        try:
+
+            def do_it():
+                if request.model_engine is not None:
+                    tokenizer_client = tiktoken.encoding_for_model(request.model_engine)
+                else:
+                    tokenizer_client = tiktoken.get_encoding(request.tokenizer)
```

Comment:
Let's just use the convention that the HELM tokenizer name will be the encoding name `openai/encoding_name` e.g. `openai/cl100k_base`. Then we can get rid of the `model_engine` field and this if-else.

Reply:
Done.